### PR TITLE
text_editing_mu4

### PR DESCRIPTION
**Pull request by vpereverzev**
Extracted the neccessary logic for text editing into the MU4

### CODE REVIEW

File: mu4/scenes/notation/view/notationviewinputcontroller.cpp [hunk 1]
```diff
@@ -148,11 +161,23 @@ void NotationViewInputController::startDragElements(ElementType etype, const QPo
...
+    }
+}
+
+void NotationViewInputController::mouseDoubleClickEvent(QMouseEvent* ev)
+{
+    Element* element = m_view->notationInteraction()->selection()->element();
+
+    if (!element) {
+        return;
     }
```
Comment (igorkorsukov) [inline mu4/scenes/notation/view/notationviewinputcontroller.cpp:178]: Maybe it’s better to add another check here that the element is text? For semantics, to make it clearer

Comment (vpereverzev) [reply mu4/scenes/notation/view/notationviewinputcontroller.cpp:181]: mm, yes, why not

File: mu4/scenes/notation/view/notationpaintview.cpp [hunk 2]
```diff
@@ -279,6 +276,15 @@ void NotationPaintView::mouseMoveEvent(QMouseEvent* ev)
     m_inputController->mouseMoveEvent(ev);
 }
 
+void NotationPaintView::mouseDoubleClickEvent(QMouseEvent* event)
+{
+    forceActiveFocus();
```
Comment (Eism) [inline mu4/scenes/notation/view/notationpaintview.cpp:281]: Why not setFocus(true) as in mousePressEvent?

Comment (vpereverzev) [reply mu4/scenes/notation/view/notationpaintview.cpp:281]: It won't work, only "forceActiveFocuse" will grab the real focus

**Pull request merged**